app.ts

```diff
@@ -0,0 +1,372 @@
+let api: string = "http://localhost:2050/";
+let isFunctionRunning = false;
+let currentPage = 1;
+let firstPage = 1
+let lastPage = firstPage + 9
+let currentFromID = 1;
+let currentToID = 20;
+let difference = 0
```

Comment:
Try to only make use of 2 global variables and give them a type

Reply:
No need to give the Type if you set the variable

Reply:
We should not have any global variables.

And you only have to set the type if they where class variables or if they were normal variables that were not assigned a value in the same line. And these variables are neither.